traits/trait_handlers.py

```diff
@@ -2757,6 +2757,7 @@ def update ( self, value ):
         if not hasattr(self, 'trait'):
             return set.update(self, value)
         try:
+            value = set(value)
```

Comment:
This already has my LGTM, but I'd rather we not create a new set unless we have to.